models/panic.py

```diff
@@ -56,26 +53,35 @@ def agent_action(agent, **kwargs):
             if DEBUG.debug:
                 print("Changing the agent's group to calm!")
             agent.has_acted = True
-            get_model(agent.exec_key).add_switch(str(agent), PANIC, CALM)
+            mdl.add_switch(str(agent), PANIC, CALM)
     return DONT_MOVE
 
 
-def start_panic(exec_key):
-    """
-    This function should be rewritten.
-    We will make a new group method called `get_rand_subset(n)`.
-    Then we will flip those agents to panicked.
-    """
-    maxPosn = panic_grps[CALM][WIDTH] * panic_grps[CALM][HEIGHT]
-    num_panic = panic_grps[PANIC][PANICKED]
-    for i in range(0, num_panic):
-        agent_posn = rand.randint(0, maxPosn)
+def get_rand_subset(n_panic, max_pos, exec_key):
```

Comment:
Cant seem to find references for this function in the code base?
Where is it supposed to be used from?

Reply:
It is the programmer's misunderstanding of what he was supposed to be doing.
I had already written the method in group.